extensions/common/vault/vault-hashicorp/src/main/java/org/eclipse/edc/vault/hashicorp/HashicorpVaultClient.java

```diff
@@ -105,76 +236,48 @@ public Result<CreateEntryResponsePayload> setSecret(@NotNull String key, @NotNul
 
         try (var response = httpClient.execute(request)) {
             if (response.isSuccessful()) {
-                var responseBody = Objects.requireNonNull(response.body()).string();
+                if (response.body() == null) {
+                    return Result.failure("Setting secret %s returned empty body".formatted(key));
```

Comment:
The secret key should not be recorded. The client knows which key it asked for and can handle it appropriately